src/App.jsx

```diff
@@ -14,33 +15,116 @@ export const goodsFromServer = [
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const ALPHABET_SORT = 'alphabet';
+const STR_LENGTH_SORT = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [goodsList, setGoodsList] = useState(goodsFromServer);
+  const [rev, setRev] = useState(false);
+  const [classActive, setClassActive] = useState('');
+  const [reverseActive, setREverseActive] = useState(true);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const getPreparedGoods = (goods, { sort, reverse }) => {
+    const preparedGoods = [...goods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (sort) {
+      return preparedGoods.sort((good1, good2) => {
+        switch (sort) {
+          case ALPHABET_SORT:
+            return good1.localeCompare(good2);
+          case STR_LENGTH_SORT:
+            return good1.length - good2.length;
+          default:
+            return 0;
+        }
+      });
+    }
+
+    if (reverse) {
+      return preparedGoods.reverse();
+    }
+
+    return preparedGoods;
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => {
+            const obj = {
+              sort: ALPHABET_SORT,
+              reverse: rev,
+            };
+
+            setGoodsList(getPreparedGoods(goodsList, obj));
+            setRev(true);
+            setClassActive(ALPHABET_SORT);
+            setREverseActive(true);
+          }}
+          type="button"
+          className={`button is-info ${classActive === ALPHABET_SORT ? '' : 'is-light'}`}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => {
+            const obj = {
+              sort: STR_LENGTH_SORT,
+              reverse: rev,
+            };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+            setGoodsList(getPreparedGoods(goodsList, obj));
+            setRev(true);
+            setClassActive(STR_LENGTH_SORT);
+            setREverseActive(true);
+          }}
+          type="button"
+          className={`button is-success ${classActive === STR_LENGTH_SORT ? '' : 'is-light'}`}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => {
+            const obj = {
+              sort: '',
+              reverse: rev,
+            };
+
+            setREverseActive(!reverseActive);
+
+            return rev ? setGoodsList(getPreparedGoods(goodsList, obj)) : null;
```

Comment:
The logic for reversing the list seems incorrect. The `reverse` property in the `obj` is always set to the current state of `rev`, which might not reflect the intended behavior. Consider toggling the `rev` state to achieve the desired reverse functionality.